### PR TITLE
Rooms

### PR DESCRIPTION
**Pull request by michel94**

### CODE REVIEW

File: api/api.py [hunk 1]
```diff
@@ -1,13 +1,41 @@
...
+    rooms[room_num] = {'users': [uid]}
+    return json.dumps({
+        "name": room_num,
+        "uid": str(uid)
+    })
+
+@app.route("/api/room/join", methods=["POST"])
+def join_room():
+    uid = uuid.uuid4()
+    print(request.json)
```
Comment (davidgomes) [inline api/api.py]: Remove.

File: api/api.py [hunk 2]
```diff
@@ -1,13 +1,41 @@
...
-@app.route("/example", methods=["GET"])
-def api_example():
+rooms = {}
+
+@app.route("/api/room", methods=["POST"])
+def create_room():
+    room_num = ("%d" % randint(0, 100)).zfill(2)
+    uid = uuid.uuid4()
+
+    rooms[room_num] = {'users': [uid]}
```
Comment (davidgomes) [inline api/api.py:15]: Get rid of ' for ".

File: api/api.py [hunk 3]
```diff
@@ -1,13 +1,41 @@
 import flask
+from flask import request
```
Comment (davidgomes) [inline api/api.py]: Get rid of this, use `flask.request`.

File: api/api.py [hunk 4]
```diff
@@ -1,13 +1,40 @@
...
+    })
+
+@app.route("/api/room/join", methods=["POST"])
+def join_room():
+    uid = uuid.uuid4()
+    
+    if request.json == None:
+        return 'Expected json data'
+    elif 'room' not in request.json:
+        return 'Expected room in json data'
```
Comment (davidgomes) [inline api/api.py:28]: Throw 400 here.

**Comment by michel94** [comment]
Too many bugs...

**Pull request closed without merging**